### PR TITLE
feat(sink): with retry

### PR DESCRIPTION
**Pull request by hekike**
Improve handling when neither offset commit nor redis sink succeeds in sink worker.

### CODE REVIEW

File: internal/sink/sink.go [hunk 1]
```diff
@@ -188,13 +189,28 @@ func (s *Sink) flush() error {
 		offsetStore.Add(message.KafkaMessage.TopicPartition)
 	}
 	offsets := offsetStore.Get()
+	var offsetCommitFailure bool
 	if len(offsets) > 0 {
-		commitedOffsets, err := s.consumer.CommitOffsets(offsets)
+		// We retry with exponential backoff as it's critical that either step #2 or #3 succeeds.
+		err := retry.Do(
```
Comment (sagikazarmark) [inline internal/sink/sink.go:195]: How long do we retry? I don't see that (or any other details about the exponential retry) here.

File: go.mod [hunk 2]
```diff
@@ -52,6 +52,7 @@ require (
 	github.com/ajg/form v1.5.1 // indirect
 	github.com/andybalholm/brotli v1.0.5 // indirect
 	github.com/apapsch/go-jsonmerge/v2 v2.0.0 // indirect
+	github.com/avast/retry-go/v4 v4.5.0
```
Comment (sagikazarmark) [inline go.mod:55]: I would have preferred using https://github.com/cenkalti/backoff as I know that better.

File: internal/sink/sink.go (hunk 1 above)
Comment (hekike) [reply internal/sink/sink.go:195]: It uses defaults currently:

```
var (
	DefaultAttempts      = [uint](https://pkg.go.dev/builtin#uint)(10)
	DefaultDelay         = 100 * [time](https://pkg.go.dev/time).[Millisecond](https://pkg.go.dev/time#Millisecond)
	DefaultMaxJitter     = 100 * [time](https://pkg.go.dev/time).[Millisecond](https://pkg.go.dev/time#Millisecond)
	DefaultOnRetry       = func(n [uint](https://pkg.go.dev/builtin#uint), err [error](https://pkg.go.dev/builtin#error)) {}
	DefaultRetryIf       = [IsRecoverable](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#IsRecoverable)
	DefaultDelayType     = [CombineDelay](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#CombineDelay)([BackOffDelay](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#BackOffDelay), [RandomDelay](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#RandomDelay))
	DefaultLastErrorOnly = [false](https://pkg.go.dev/builtin#false)
	DefaultContext       = [context](https://pkg.go.dev/context).[Background](https://pkg.go.dev/context#Background)()
)
```

Comment (hekike) [reply internal/sink/sink.go:195]: It's a backoff with jitter

File: go.mod (hunk 2 above)
Comment (hekike) [reply go.mod:55]: Discussed with DM, giving a try to https://github.com/avast/retry-go

**Pull request merged**